### PR TITLE
Add a Ruckig jerk-limited smoothing plugin, mostly for Servo

### PR DESCRIPTION
**Pull request by AndyZe**
This builds off of PR #676, so please review that one first.

Add a Ruckig jerk-limited smoothing plugin. Right now, this can easily be used in Servo. In the future, it might be useful in hybrid planning.

(Sorry for the flood of smoothing-related PR's recently. This is being driven by client work.)

Here are videos of the Ruckig filter (top) and the Butterworth filter. They look pretty much the same to my eye.

https://user-images.githubusercontent.com/11284393/133804964-abc72f66-9c05-432a-a58f-34c28e5ea12e.mp4

https://user-images.githubusercontent.com/11284393/133804979-188f8027-b478-4512-8444-1533719151dc.mp4

@destogl

### CODE REVIEW

**Review by nbbrooks — COMMENTED** [review]
Need to enforce smoothing in the suddenHalt functions. Right now they try and immediately halt in 1ns.

File: moveit_core/single_waypt_smoothing_plugins/include/moveit/single_waypt_smoothing_plugins/ruckig_filter.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                  double timestep) override;
+
+  bool doSmoothing(std::vector<double>& /*unused*/, std::vector<double>& velocity_vector) override;
+
+  bool reset(const std::vector<double>& joint_positions) override;
+
+private:
+  size_t num_dof_;
+  rclcpp::Node::SharedPtr node_;
+  std::shared_ptr<ruckig::Ruckig<0, true /*debug*/>> ruckig_;
```
Comment (AndyZe) [inline moveit_core/single_waypt_smoothing_plugins/include/moveit/single_waypt_smoothing_plugins/ruckig_filter.h:64]: TODO: delete the ruckig debug flag

File: moveit_core/single_waypt_smoothing_plugins/src/ruckig_filter.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  if (ruckig_)
+  {
+    result = ruckig_->update(*ruckig_input_, *ruckig_output_);
+  }
+  else
+  {
+    return false;
+  }
+
+  return result == ruckig::Result::Finished;
```
Comment (AndyZe) [inline moveit_core/single_waypt_smoothing_plugins/src/ruckig_filter.cpp:100]: TODO: I think this should be ::Working

**Comment by AndyZe** [comment]
> Need to enforce smoothing in the suddenHalt functions. Right now they try and immediately halt in 1ns.

I added that in #676 so it works for all plugins, not just the Ruckig one.  :+1:

**Comment by AndyZe** [comment]
Rebasing this one got tricky. Will open a new PR

**Pull request closed without merging**